src/ui/public/kbn_top_nav/kbn_top_nav_controller.js

```diff
@@ -53,13 +53,18 @@ export default function ($compile) {
 
     // apply the defaults to individual options
     _applyOptDefault(opt = {}) {
-      return defaults({}, opt, {
+      const defaultedOpt = defaults({}, opt, {
         label: capitalize(opt.key),
         hasFunction: !!opt.run,
         description: opt.run ? opt.key : `Toggle ${opt.key} view`,
-        hideButton: !!opt.hideButton,
-        run: (item) => this.toggle(item.key)
+        run: (item) => !item.disableButton && this.toggle(item.key)
       });
+
+      defaultedOpt.hideButton = result(opt, 'hideButton', false);
+      defaultedOpt.disableButton = result(opt, 'disableButton', false);
+      defaultedOpt.tooltip = result(opt, 'tooltip', '');
```

Comment:
I think we don't want them to always be functions. For more static uses (where the nav state isn't being toggled dynamically), it would be nice for the caller to just specify a static, scalar value. The function is only if they want the value to be dynamic.